compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -5945,12 +5918,9 @@ object Types {
             case Range(lo, hi) => range(bound.bounds.lo, bound.bounds.hi)
             case _ => tp.derivedMatchType(bound, scrutinee, cases)
 
-    override protected def derivedSkolemType(tp: SkolemType, info: Type): Type = info match {
-      case Range(lo, hi) =>
-        range(tp.derivedSkolemType(lo), tp.derivedSkolemType(hi))
-      case _ =>
-        tp.derivedSkolemType(info)
-    }
+    override protected def derivedSkolemType(tp: SkolemType, info: Type): Type =
+      if info eq tp.info then tp
```

Comment:
`Range` is special-cased in situations where we need to rewrap its components to construct a new Range, but here we don't need to: we always want to construct a `range(Nothing, info)` and `range` itself correctly handles the situation where `info` is a `Range` (it will replace it by its upper-bound).